mxcube3/core/components/sampleview.py

```diff
@@ -39,6 +39,27 @@ def __init__(self, app, config):
             HWR.beamline.collect, HWR.beamline.diffractometer, HWR.beamline.sample_view
         )
 
+        HWR.beamline.sample_view.connect("shapesChanged", self._emit_shapes_updated)
+
+        zoom_motor = HWR.beamline.diffractometer.get_object_by_role("zoom")
+
+        if zoom_motor:
+            zoom_motor.connect("stateChanged", self._zoom_changed)
+
+    def _zoom_changed(self, *args, **kwargs):
+        ppm = HWR.beamline.diffractometer.get_pixels_per_mm()
+        self.app.server.emit("update_pixels_per_mm", {
+                             "pixelsPerMm": ppm}, namespace="/hwr")
+
+    def _emit_shapes_updated(self):
+        shape_dict = {}
+
+        for shape in HWR.beamline.sample_view.get_shapes():
+            _s = to_camel(shape.as_dict())
```

Comment:
Nothing wrong with this, but why not just do:
    shape_dict[shape.id] = to_camel(shape.as_dict())
?